typespec-extension/src/code-model-builder.ts

```diff
@@ -1022,9 +1032,10 @@ export class CodeModelBuilder {
         (hasScalarAsBase(param.param.type, "utcDateTime") || hasScalarAsBase(param.param.type, "offsetDateTime"))
       ) {
         // utcDateTime in header maps to rfc7231
-        schema = this.processDateTimeSchema(param.param.type, param.param.name, true);
+        schema = this.processDateTimeSchemaFromSdkType(sdkType as SdkDatetimeType, param.param.name, true);
```

Comment:
Why use `as`? What if this is not SdkDatetimeType?

Do not use `as` unless there is no other way. Here a check and fallback should work fine.

Reply:
BTW, in processDateTimeSchemaFromSdkType you actually didn't do anything about SdkDatetimeType